rimage/lazy_encoded.go

```diff
@@ -99,28 +127,48 @@ func (lei *LazyEncodedImage) RawData() []byte {
 }
 
 // DecodedImage returns the decoded image.
+//
+// It is recommended to call DecodeImage and check for errors before using this method.
 func (lei *LazyEncodedImage) DecodedImage() image.Image {
-	lei.decode()
+	err := lei.DecodeImage()
+	if err != nil {
+		panic(err)
+	}
 	return lei.decodedImage
 }
 
 // ColorModel returns the Image's color model.
+//
+// It is recommended to call DecodeConfig and check for errors before using this method.
 func (lei *LazyEncodedImage) ColorModel() color.Model {
-	lei.decodeConfig()
+	err := lei.DecodeConfig()
+	if err != nil {
+		panic(err)
+	}
 	return lei.colorModel
 }
 
 // Bounds returns the domain for which At can return non-zero color.
 // The bounds do not necessarily contain the point (0, 0).
+//
+// It is recommended to call DecodeConfig and check for errors before using this method.
 func (lei *LazyEncodedImage) Bounds() image.Rectangle {
-	lei.decodeConfig()
+	err := lei.DecodeConfig()
+	if err != nil {
+		panic(err)
```

Comment:
Still a little concerned about having panics here. Wondering if it would be better to have error cases in the helpers `Bounds`, `ColorModel`, and `At` instead of relying on the consumer to check for errors from `DecodeImage` beforehand.

Reply:
Good point. However, the `Bounds` `ColorModel` and `At` methods are of the `image.Image` interface and cannot be modified. Nick raised an idea of, in addition to what we have alr, add a bunch of `-Safe` suffixed versions of the panicking functions. This is nice, but these methods also won't show up unless we type assert `(*LazyEncodedImage)`

Reply:
We kinda just have to live under the assumption that some of these `image.Image` we are passing around are secret lazy images and may explode with a panic 😢 

An questionable design choice we must live with.

Reply:
Yeah makes sense. We have to conform to image.Image interface and we cant just return nil struct beacuse of golang...

Reply:
What about just returning an empty struct in the error case instead of panicking?

Reply:
That may create a very opaque failure state

Reply:
> add a bunch of -Safe suffixed versions of the panicking functions. This is nice, but these methods also won't show up unless we type assert (*LazyEncodedImage)

These were added btw @nicksanford